atlas-spec/mpc-engine/src/primitives/mac.rs

```diff
@@ -1,8 +1,65 @@
 //! This module defines an information theoretic MAC for authenticating bits.
 
-use crate::COMPUTATIONAL_SECURITY;
+use hacspec_lib::Randomness;
+
+use crate::{Error, COMPUTATIONAL_SECURITY};
+
+/// The length in bytes of an information theoretic MAC, and of the MAC key.
+pub const MAC_LENGTH: usize = COMPUTATIONAL_SECURITY;
 
 /// A MAC on a bit.
-pub type Mac = [u8; COMPUTATIONAL_SECURITY];
+pub type Mac = [u8; MAC_LENGTH];
 /// A MAC key for authenticating a bit to another party.
-pub type MacKey = [u8; COMPUTATIONAL_SECURITY];
+pub type MacKey = [u8; MAC_LENGTH];
+
+/// Generate a fresh MAC key.
+pub fn generate_mac_key(entropy: &mut Randomness) -> Result<MacKey, Error> {
+    let k: [u8; MAC_LENGTH] = entropy
+        .bytes(MAC_LENGTH)?
+        .try_into()
+        .map_err(|_| Error::OtherError)?;
+    Ok(k)
+}
+
+/// Authenticate a bit using the global MAC key.
+pub fn mac(
+    bit: &bool,
+    global_key: &MacKey,
+    entropy: &mut Randomness,
+) -> Result<(Mac, MacKey), Error> {
+    let key: [u8; MAC_LENGTH] = generate_mac_key(entropy)?;
+
+    let mut mac = [0u8; MAC_LENGTH];
+    for idx in 0..mac.len() {
+        mac[idx] = key[idx] ^ (if *bit { global_key[idx] } else { 0x00 });
```

Comment:
This looks weird. Why is the if needed here?

Reply:
You're right, it's better expressed casting `*bit` to `u8`.